db/migrate/20220718083657_create_wallets.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class CreateWallets < ActiveRecord::Migration[7.0]
+  def change
+    create_table :wallets, id: :uuid do |t|
+      t.references :customer, type: :uuid, null: false, foreign_key: true, index: true
+
+      t.integer :status, null: false
+      t.string :currency, null: false
+
+      t.string :name
+      t.string :rate_amount, null: false
```

Comment:
Why we're not using `bigint` here?

Reply:
we'll use `BigDecimal` since we want to support 5 decimals, it's the best way to avoid strange rouding or behaviours

Reply:
we store `BigDecimal` as `string`! It's the best way we found to avoid issues, do you have more exp with the `BigDecimal` behaviour?